operator/watchers/node_taint.go

```diff
@@ -127,74 +145,23 @@ func ciliumPodsWatcher(wg *sync.WaitGroup, clientset k8sClient.Clientset, stopCh
 		0,
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
-				key, _ := queueKeyFunc(obj)
-				ciliumQueue.Add(key)
+				ciliumPodHandler(obj, queue, logger)
 			},
 			UpdateFunc: func(_, newObj interface{}) {
-				key, _ := queueKeyFunc(newObj)
-				ciliumQueue.Add(key)
+				ciliumPodHandler(newObj, queue, logger)
```

Comment:
Any sense in eliding changes here if the interesting state (schedule,d & readiness) are unchanged between old and new pods?

Reply:
Makes sense, although I would do it in a separate PR as the next optimization.